src/Rules/Configure.hs

```diff
@@ -12,7 +12,7 @@ import Utilities
 
 configureRules :: Rules ()
 configureRules = do
-    [configFile, "settings", configH, "compiler/ghc.cabal"] &%> \outs -> do
```

Comment:
Why has `compiler/ghc.cabal` been removed from here? I guess we are still generating it in `configure`?

Reply:
Just because I added it at some point, thinking it would fix the problem I was seeing, but it didn't. It does not accomplish anything and I'm also thinking we just don't want to list all the files that `./configure` produces... do we? =)

Reply:
I think we should strive to list all dependencies here! Isn't it nice that Hadrian can automatically reconfigure if you edit `ghc.cabal.in`?

Reply:
Yes, indeed. Maybe we can get away with a "fancy" pattern, `**.in` or something?

Reply:
Wait, no, we can't do that, because the `.in` files are the inputs, not outputs. Hmm, this might be a little painful...

``` sh
$ find . -name '*.in'
./_build/stage1/libffi/build/man/Makefile.in
./_build/stage1/libffi/build/include/ffi.h.in
./_build/stage1/libffi/build/include/Makefile.in
./_build/stage1/libffi/build/libffi.pc.in
./_build/stage1/libffi/build/doc/Makefile.in
./_build/stage1/libffi/build/libffi.map.in
./_build/stage1/libffi/build/fficonfig.h.in
./_build/stage1/libffi/build/Makefile.in
./_build/stage1/libffi/build/testsuite/Makefile.in
./nofib/real/scs/runtime_files/inverter.in
./libraries/terminfo/terminfo.buildinfo.in
./libraries/Cabal/cabal-install/tests/IntegrationTests2/exception/no-pkg/empty.in
./libraries/Cabal/cabal-testsuite/PackageTests/NewBuild/MonitorCabalFiles/q/q-fixed.cabal.in
./libraries/Cabal/cabal-testsuite/PackageTests/NewBuild/MonitorCabalFiles/q/q-broken.cabal.in
./libraries/Cabal/cabal-testsuite/PackageTests/Configure/include/HsZlibConfig.h.in
./libraries/Cabal/cabal-testsuite/PackageTests/Configure/zlib.buildinfo.in
./libraries/time/lib/include/HsTimeConfig.h.in
./libraries/ghci/ghci.cabal.in
./libraries/prologue.txt.in
./libraries/ghc-boot/ghc-boot.cabal.in
./libraries/base/include/HsBaseConfig.h.in
./libraries/base/include/EventConfig.h.in
./libraries/base/tests/IO/decodingerror002.in
./libraries/base/tests/IO/hGetPosn001.in
./libraries/base/tests/IO/hSeek001.in
./libraries/base/tests/IO/hSetEncoding001.in
./libraries/base/base.buildinfo.in
./libraries/integer-gmp/include/HsIntegerGmp.h.in
./libraries/integer-gmp/config.mk.in
./libraries/integer-gmp/integer-gmp.buildinfo.in
./libraries/ghc-boot-th/ghc-boot-th.cabal.in
./libraries/unix/include/HsUnixConfig.h.in
./libraries/unix/unix.buildinfo.in
./libraries/ghc-prim/ghc-prim.buildinfo.in
./libraries/process/include/HsProcessConfig.h.in
./libraries/directory/HsDirectoryConfig.h.in
./rts/package.conf.in
./rts/rts.cabal.in
./distrib/configure.ac.in
./docs/users_guide/ghc_config.py.in
./docs/index.html.in
./settings.in
./hadrian/dist-newstyle/src/happy-1.19.9/doc/config.mk.in
./hadrian/dist-newstyle/src/happy-1.19.9/doc/happy.1.in
./hadrian/cfg/system.config.in
./compiler/ghc.cabal.in
./utils/runghc/runghc.cabal.in
./utils/gen-dll/gen-dll.cabal.in
./ghc/ghc-bin.cabal.in
./testsuite/tests/backpack/cabal/bkpcabal04/q/A/B.hsig.in
./testsuite/tests/backpack/cabal/bkpcabal04/q/B.hsig.in
./mk/project.mk.in
./mk/install.mk.in
./mk/config.h.in
./mk/config.mk.in
```

If we leave out all the unnecessary ones, we're left with:

```
$ find . -name '*.in'
./nofib/real/scs/runtime_files/inverter.in
./libraries/terminfo/terminfo.buildinfo.in
./libraries/Cabal/cabal-install/tests/IntegrationTests2/exception/no-pkg/empty.in
./libraries/Cabal/cabal-testsuite/PackageTests/NewBuild/MonitorCabalFiles/q/q-fixed.cabal.in
./libraries/Cabal/cabal-testsuite/PackageTests/NewBuild/MonitorCabalFiles/q/q-broken.cabal.in
./libraries/Cabal/cabal-testsuite/PackageTests/Configure/include/HsZlibConfig.h.in
./libraries/Cabal/cabal-testsuite/PackageTests/Configure/zlib.buildinfo.in
./libraries/time/lib/include/HsTimeConfig.h.in
./libraries/ghci/ghci.cabal.in
./libraries/prologue.txt.in
./libraries/ghc-boot/ghc-boot.cabal.in
./libraries/base/include/HsBaseConfig.h.in
./libraries/base/include/EventConfig.h.in
./libraries/base/base.buildinfo.in
./libraries/integer-gmp/include/HsIntegerGmp.h.in
./libraries/integer-gmp/integer-gmp.buildinfo.in
./libraries/ghc-boot-th/ghc-boot-th.cabal.in
./libraries/unix/include/HsUnixConfig.h.in
./libraries/unix/unix.buildinfo.in
./libraries/ghc-prim/ghc-prim.buildinfo.in
./libraries/process/include/HsProcessConfig.h.in
./libraries/directory/HsDirectoryConfig.h.in
./rts/package.conf.in
./rts/rts.cabal.in
./distrib/configure.ac.in
./docs/users_guide/ghc_config.py.in
./docs/index.html.in
./settings.in
./hadrian/cfg/system.config.in
./compiler/ghc.cabal.in
./utils/runghc/runghc.cabal.in
./utils/gen-dll/gen-dll.cabal.in
./ghc/ghc-bin.cabal.in
./mk/config.h.in
```

if I'm not mistaken? Shall we add all those files, hoping the list doesn't change too often?

Reply:
OK, done in latest commit :-)